test/process_stealing_test.ml

```diff
@@ -0,0 +1,33 @@
+[@@@warning "-8"]
+
+open Riot
+
+type Message.t += A
+
+let rec loop pid msg n =
+  if n = 0 then send pid msg
+  else (
+    yield ();
+    loop pid msg (n - 1))
+
+let main () =
+  Riot.run @@ fun () ->
+  let this = self () in
+
+  let _ =
+    spawn_pinned (fun () ->
+        yield ();
+        loop this A 100)
+  in
+  let _low_priority_process =
+    spawn_pinned (fun () ->
+        process_flag (Priority Low);
+        yield ();
+        loop this A 100)
+  in
+  (* I think that here, the low priority process should be stolen by another
+     scheduler (and it's SID will not equal "00"). However, I'm not sure how to
+     check for this (or if it's working). *)
+  ()
```

Comment:
For the exit condition, I think we should be able to get process information (like the scheduler id) with a function, so we can compare for both process and it should be different.

I get a feeling that this will only happen as long as:
* A lives longer than B
* A is more resource intensive, so that it hogs its scheduler

What do you think?